File: ButtonCore/src/main/java/buttondevteam/lib/TBMCCommandPreprocessEvent.java
package buttondevteam.lib;

import lombok.Getter;
import lombok.Setter;
import org.bukkit.command.CommandSender;
import org.bukkit.event.Cancellable;
import org.bukkit.event.Event;
import org.bukkit.event.HandlerList;

/**
 * Can be used to change or handle commands before they're sent.
 * <b>Called on using player, console and Discord commands.</b>
 *
 * @author NorbiPeti
 */
@Getter
public class TBMCCommandPreprocessEvent extends Event implements Cancellable {
	private static final HandlerList handlers = new HandlerList();

	private final CommandSender sender;
	@Setter
	private String message;
	@Setter
	private boolean cancelled;

	public TBMCCommandPreprocessEvent(CommandSender sender, String message) {
		this.sender = sender;
		this.message = message; //TODO: Actually call from Discord as well
	}

	@Override
	public HandlerList getHandlers() {
		return handlers;
	}

	public static HandlerList getHandlerList() {
		return handlers;
	}
}

File: Chroma-Core/src/main/java/buttondevteam/lib/TBMCCommandPreprocessEvent.kt
package buttondevteam.lib

import buttondevteam.core.component.channel.Channel
import buttondevteam.lib.player.ChromaGamerBase
import org.bukkit.command.CommandSender
import org.bukkit.event.Cancellable
import org.bukkit.event.Event
import org.bukkit.event.HandlerList

/**
 * Can be used to change or handle commands before they're sent.
 * **Called on using player, console and Discord commands.**
 *
 * @author NorbiPeti
 */
class TBMCCommandPreprocessEvent(
    val sender: ChromaGamerBase,
    val channel: Channel,
    val message: String,
    val permCheck: CommandSender
) : Event(), Cancellable {
    private var cancelled = false
    override fun getHandlers(): HandlerList {
        return handlerList
    }

    override fun isCancelled() = cancelled

    override fun setCancelled(cancelled: Boolean) = run { this.cancelled = cancelled }

    companion object {
        val handlerList = HandlerList()
    }
}